viewing_party/party.py

```diff
@@ -1,23 +1,122 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if not title or not genre or not rating:
+        return None
+    else:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for i in range(len(user_data["watchlist"])):
+        if (user_data["watchlist"][i]["title"]) == title:
+            watched_movie = (user_data["watchlist"].pop(i))
+            user_data["watched"].append(watched_movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    sum_of_ratings = 0
+    if not user_data["watched"]:
+        return 0.0
+    else:

```

Comment:
Great use of the guard clause pattern again! Similar to my above comment, since the guard clause returns early and we again definitely want what's in this `else` block to execute, we can omit the `else` and bring the contents of the block in alignment with the indentation of the `if` statement.